developer/js/source/lexical-model-compiler/lexical-model.ts

```diff
@@ -45,6 +45,26 @@ interface WordBreakerSpec {
  */
 type SimpleWordBreakerSpec = 'default' | 'ascii' | WordBreakingFunction;
 
+/**
+ * Simplifies input text to facilitate finding entries within a lexical model's
+ * lexicon.
+ * @since 11.0
+ */
+type SimpleWordformToKeySpec = (term: string) => string;
+
+/**
+ * Simplifies input text to facilitate finding entries within a lexical model's
+ * lexicon, using the model's `applyCasing` function to assist in the keying process.
+ * @since 14.0
+ */
+ type CasedWordformToKeySpec = (term: string, applyCasing?: CasingFunction) => string;
```

Comment:
Make `applyCasing` a required argument, else this is just an extension of `SimpleWorformToKeySpec`.

```suggestion
 type CasedWordformToKeySpec = (term: string, applyCasing: CasingFunction) => string;
```

Reply:
If a developer chooses to write their own custom `searchTermToKey` function for a model that uses casing behaviors, this would force the model developer's implementation to accept the `applyCasing` parameter.  Admittedly, that's _probably_ wise, but is it valid to _force_ this upon them?